src/DIPS.Xamarin.UI/Controls/TrendGraph/TrendGraph.xaml.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections;
+using Xamarin.Forms;
+using DIPS.Xamarin.UI.Extensions;
+using System.Collections.Specialized;
+
+namespace DIPS.Xamarin.UI.Controls.TrendGraph
+{
+    public partial class TrendGraph : ContentView
+    {
+        private VisualElement? m_previousParent;
+        public TrendGraph()
+        {
+            HorizontalOptions = LayoutOptions.FillAndExpand;
+            VerticalOptions = LayoutOptions.FillAndExpand;
+            SizeChanged += (s, e) => StartRedraw(this, EventArgs.Empty);
+            InitializeComponent();
+        }
+
+        protected override void OnParentSet()
+        {
+            var parentSize = ((VisualElement)Parent).Width;
+            if (m_previousParent != null)
+            {
+                m_previousParent.SizeChanged -= StartRedraw;
+                m_previousParent = null;
+            }
+
+            if (Parent is VisualElement element)
+            {
+                element.SizeChanged += StartRedraw;
+                m_previousParent = element;
+                StartRedraw(this, EventArgs.Empty);
+            }
+
+            base.OnParentSet();
+        }
+
+        private void StartRedraw(object _, EventArgs __)
+        {
+            Redraw();
+        }
+
+        private void Redraw()
+        {
+            graphContainer.Children.Clear();
+            if (MinValue >= MaxValue ||
```

Comment:
Maybe we can throw for each of this cases? To not end up with nothing that is visible and no way of knowing what the consumers are missing?
I don't know if it is a good solution, but something to think about